packages/polling-controller/src/PollingController.ts

```diff
@@ -112,24 +122,29 @@ function PollingControllerMixin<TBase extends Constructor>(Base: TBase) {
      * Executes the poll for a networkClientId
      *
      * @param networkClientId - The networkClientId to execute the poll for
+     * @param options - The options passed to startPollingByNetworkClientId
      */
-    abstract executePoll(networkClientId: NetworkClientId): Promise<void>;
+    abstract executePoll(
+      networkClientId: NetworkClientId,
+      options: any,
```

Comment:
should options be an object or a generic<T>?

Reply:
I'm wondering that as well.

Can you talk about how these options would get used? I'm seeing them being passed to `executePoll`, but from there how would they be used?

Reply:
one example for it to get used is the [NFTDetectionController PR](https://github.com/MetaMask/core/pull/1758/files#top).

a consumer of `NFTDetectionController` would start polling like this:

```
nftDetectionController.startPollingByNetworkClientId('mainnet', {address: <some selected address> })
```

Reply:
Ah, gotcha. So it sounds like this could really be anything that a consumer want to pass in order to scope the polling token keys.